### PR TITLE
0.0.9 update

### PR DESCRIPTION
**Pull request by ViridianMelody**
Upstream repo: https://github.com/dtkav/swagger_ui_bundle/tree/master
Jira Issue: https://anaconda.atlassian.net/browse/PKG-114
Source diff: https://github.com/dtkav/swagger_ui_bundle/compare/v0.0.8...v0.0.9

### CODE REVIEW

**Review by Arishamays1 — APPROVED** [review]
LGTM!

File: abs.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+channels:
```
Comment (ryanskeith) [inline abs.yaml]: Please remove abs.yam before merging.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -21,20 +21,30 @@ requirements:
     - python
     - flake8
```
Comment (ryanskeith) [inline recipe/meta.yaml:22]: If you updated flake8 because of this, you should probably note the version necessary to make it work here.

**Review by ryanskeith — CHANGES_REQUESTED** [review]
Look great.  Please see comments before merging.

**Review by varlackc — COMMENTED** [review]
While the earliest version of `python` that we support is `3.7`, the earliest `python` version supported in the upstream is `3.4`
https://github.com/dtkav/swagger_ui_bundle/blob/master/setup.py#L76

We might want to remove the quotes on the `home_url`
https://github.com/AnacondaRecipes/swagger-ui-bundle-feedstock/blob/234cbe59f69ec8003428e8ed1fb91aa542c657cd/recipe/meta.yaml#L38

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
 
 package:
   name: "{{ name|lower }}"
   version: "{{ version }}"
 
 source:
-  url: "https://github.com/dtkav/swagger_ui_bundle/archive/v{{ version }}.tar.gz"
-  sha256: "1d49f77f53ac6b25d7413bb9239893bd9eff1e7bb570dc3c57b387a345f3b5cd"
+  url: "https://github.com/dtkav/swagger_ui_bundle/archive/refs/tags/v{{ version }}.tar.gz"
+  sha256: "abebdbb77a7ffa91e46caca35212f5bd39028db9b0a45416b51215c5e7a1b3df"
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: Please remove quotes symbols `"`

File: recipe/meta.yaml [hunk 4]
```diff
@@ -21,20 +21,30 @@ requirements:
     - python
     - flake8
     - pytest-runner
+    - wheel
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:24]: Also, missing setuptools, see https://github.com/dtkav/swagger_ui_bundle/blob/0657580c653ae1cb3829b28a72607712addd7019/setup.py#L8

File: recipe/meta.yaml [hunk 5]
```diff
@@ -21,20 +21,30 @@ requirements:
...
 test:
   imports:
     - swagger_ui_bundle
+  requires:
+    - pip
+  commands:
+    - pip check
 
 about:
   home: "https://github.com/dtkav/swagger_ui_bundle"
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: Please remove quotes `"`

File: recipe/meta.yaml [hunk 6]
```diff
@@ -21,20 +21,30 @@ requirements:
...
-  license: "Apache Version 2.0"
+  license: Apache-2.0
   license_family: "APACHE"
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: Please remove quotes `"`

File: recipe/meta.yaml [hunk 7]
```diff
@@ -21,20 +21,30 @@ requirements:
...
   license_file: "LICENSE"
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: quotes `"` are redundant

**Comment by skupr-anaconda** [comment]
@ViridianMelody It seems like `ERROR: Could not find a version that satisfies the requirement importlib-metadata<4.3 (from versions: none)` is caused by different versions of `importlib-metadata` for different `python` versions. In python 3.10 and 3.11 it's already in the standard library.
Try if this will help https://github.com/AnacondaRecipes/spyder-feedstock/blob/61d8f34cc93d0dcb8289c6fc0a4b097e4f7e09e6/recipe/meta.yaml#L71-L73

**Comment by skupr-anaconda** [comment]
It looks like Prefect has issues on win64 `PermissionError: [WinError 5] Access is denied: 'C:\\Users\\builder-admin\\AppData\\Local\\Temp\\pip-pip-egg-info-cfmruek1\\swagger_ui_bundle.egg-info\\dependency_links.txt'`

File: abs.yaml (hunk 1 above)
Comment (ViridianMelody) [reply abs.yaml]: Done!

File: recipe/meta.yaml (hunk 2 above)
Comment (ViridianMelody) [reply recipe/meta.yaml:22]: Ended up not updating flake8 for now.

File: recipe/meta.yaml [hunk 8]
```diff
@@ -21,20 +21,35 @@ requirements:
     - python
     - flake8
     - pytest-runner
+    - wheel
+    - setuptools
+    # This is here to workaround an inconsistency with flake8
+    # See: https://github.com/conda-forge/flake8-feedstock/issues/52
+    - importlib-metadata <4.3  # [py<38]
+    - mccabe >=0.6.0,<0.7.0
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:29]: There was an inconsistency in the v0.6.1 artifacts on different platforms.
mccabe 0.7.0 is a noarch python package. It has minimal changes, see https://github.com/PyCQA/mccabe/compare/0.6.1...0.7.0. You can safely use it. That's why the pinning `mccabe >=0.6.0,<0.8.0` is OK.

Comment (ViridianMelody) [reply recipe/meta.yaml:29]: Using `mccabe >=0.6.0,<0.8.0` causes the build to fail due to the mismatch with `flake8`'s metadata. `flake8` itself will build (as long as `pip check` is disabled), but not anything that depends on it.

Comment (skupr-anaconda) [reply recipe/meta.yaml:29]: Yeah, you can just comment `pip check` to disable it

Comment (ViridianMelody) [reply recipe/meta.yaml:29]: Here are the errors I was getting when attempting to build `swagger-ui-bundle` without modifying the `mccabe` pinnings:
```
ERROR: Could not find a version that satisfies the requirement mccabe<0.7.0,>=0.6.0 (from versions: none)
ERROR: No matching distribution found for mccabe<0.7.0,>=0.6.0
```

Comment (AndrewVallette) [reply recipe/meta.yaml:29]: @skupr-anaconda You are correct that there are minimal changes in `0.7.0`, but going with `<0.8.0` means that future versions like `0.7.5` or `0.7.10` could potentially be used. And we do not yet know what changes would exist.

I believe the safest choice is to use the `<0.7.0` pinning that Melody is using. This is aligned with what upstream is doing.

I should also ask, what are the inconsistencies that exist in `0.6.1`? Is it something that should truly cause us to use `0.7.0`? If so, using `<=0.7.0` is another option as well.

I'd prefer to stay with the pinnings that upstream uses, and not skip `pip check`, unless there is a specific reason.

Comment (skupr-anaconda) [reply recipe/meta.yaml:29]: @AndrewVallette Yeah, you are right, `<=0.7.0` will be the better pinning. It was a workaround to build flake8. At that time there were missing packages on some platforms. If now we can build with 0.6.1 for all python versions and all platforms we can try it

**Review by ryanskeith — APPROVED** [review]
LGTM

**Pull request merged**